jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/JmxScraper.java

```diff
@@ -233,10 +229,39 @@ private static Properties loadPropertiesFromPath(String path) throws InvalidArgu
   }
 
   private JmxScraper(
-      JmxConnectorBuilder client, JmxMetricInsight service, JmxScraperConfig config) {
+      JmxConnectorBuilder client, OpenTelemetry openTelemetry, JmxScraperConfig config) {
     this.client = client;
-    this.service = service;
-    this.config = config;
+    this.jmxTelemetry = createJmxTelemetry(openTelemetry, config);
+  }
+
+  private static JmxTelemetry createJmxTelemetry(
+      OpenTelemetry openTelemetry, JmxScraperConfig config) {
+
+    JmxTelemetryBuilder builder = JmxTelemetry.builder(openTelemetry);
+    builder.beanDiscoveryDelay(config.getSamplingInterval());
+
+    // Unfortunately we can't use the convenient 'addClassPathRules' here as it does not yet
+    // allow to customize the path of the yaml resources in classpath.
+    // config.getTargetSystems().forEach(builder::addClassPathRules);
+    //
+    // As a temporary workaround we load configuration through temporary files and register them
+    // as if they were custom rules.
+    config
+        .getTargetSystems()
+        .forEach(
+            system -> {
+              try (InputStream input = config.getTargetSystemYaml(system)) {
+                Path tempFile = Files.createTempFile("jmx-scraper-" + system, ".yaml");
+                Files.copy(input, tempFile, StandardCopyOption.REPLACE_EXISTING);
+                builder.addCustomRules(tempFile);
+                Files.delete(tempFile);
```

Comment:
Temp file will be removed even if exception happen during file copy / adding rules
```suggestion
                try {
                  Files.copy(input, tempFile, StandardCopyOption.REPLACE_EXISTING);
                  builder.addCustomRules(tempFile);
                } finally {
                  Files.delete(tempFile);
                }
```